### PR TITLE
Add CLI commands to manage LDAP authentication source

### PR DESCRIPTION
**Pull request by ngourdon**
Add 4 CLI commands:
- add-ldap: Add new LDAP (via Bind DN) authentication source
- update-ldap: Update existing LDAP (via Bind DN) authentication source
- add-ldap-simple: Add new LDAP (simple auth) authentication source 
- update-ldap-simple: Update existing LDAP (simple auth) authentication source

#5726

### CODE REVIEW

File: cmd/admin_auth_ldap.go [hunk 1]
```diff
@@ -0,0 +1,387 @@
+// Copyright 2016 The Gogs Authors. All rights reserved.
```
Comment (techknowlogick) [inline cmd/admin_auth_ldap.go]: Gogs copyright isn’t needed in this file :)

**Comment by ngourdon** [comment]
Gogs copyright is removed.
I also updated func `parseLoginSource` which wasn't right

File: cmd/admin_auth_ldap.go [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+
+func (a *authService) initLdapCommand(c *cli.Context) error {
+	if c.IsSet("config") {
+		setting.CustomConf = c.String("config")
+	}
+
+	return a.initDB()
+}
+
+// getLoginSource gets the login source by its id define in command line flags.
```
Comment (adelowo) [inline cmd/admin_auth_ldap.go]:
```suggestion
// getLoginSource gets the login source by its id defined in the command line flags.
```

**Comment by ngourdon** [comment]
Is there anything else needed for approval?

File: cmd/admin_auth_ldap.go [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+	}
+)
+
+var (
+	commonLdapCLIFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "config, c",
+			Value: "custom/conf/app.ini",
+			Usage: "Custom configuration file path.",
+		},
```
Comment (zeripath) [inline cmd/admin_auth_ldap.go]: You no longer need to add config options by default.

Comment (ngourdon) [reply cmd/admin_auth_ldap.go]: @zeripath Thanks for pointing this out.

**Comment by ngourdon** [comment]
@zeripath config flag removed

**Comment by zeripath** [comment]
@ngourdon sorry I didn't get a chance to properly look at this this evening. I'll try to review tomorrow evening UK time.

**Review by zeripath — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**